tests/test_synced_collection.py

```diff
@@ -0,0 +1,506 @@
+        assert len(sd) == 0
+        with pytest.raises(KeyError):
+            sd[key]
+        sd[key] = d
+        assert len(sd) == 1
+        assert sd[key] == d
+        del sd['delete']
+        assert len(sd) == 0
+        with pytest.raises(KeyError):
+            sd[key]
+
+    def test_update(self, synced_collection, testdata):
+        sd = synced_collection
+        key = 'update'
+        d = {key: testdata}
+        sd.update(d)
+        assert len(sd) == 1
+        assert sd[key] == d[key]
+
+    def test_pop(self, synced_collection, testdata):
+        sd = synced_collection
+        key = 'pop'
+        d = testdata
+        sd[key] = d
+        assert len(sd) == 1
+        assert sd[key] == d
+        d1 = sd.pop(key)
+        assert len(sd) == 0
+        assert d == d1
+        with pytest.raises(KeyError):
+            sd[key]
+        d2 = sd.pop(key, 'default')
+        assert len(sd) == 0
+        assert d2 == 'default'
+
+    def test_popitem(self, synced_collection, testdata):
+        sd = synced_collection
+        key = 'pop'
+        d = testdata
+        sd[key] = d
+        assert len(sd) == 1
+        assert sd[key] == d
+        key1, d1 = sd.popitem()
+        assert len(sd) == 0
+        assert key == key1
+        assert d == d1
+        with pytest.raises(KeyError):
+            sd[key]
+
+    def test_reset(self, synced_collection, testdata):
+        sd = synced_collection
+        key = 'reset'
+        d = testdata
+        sd[key] = d
+        assert len(sd) == 1
+        assert sd[key] == d
+        sd.reset()
+        assert len(sd) == 0
+        d1 = testdata
+        sd.reset({'reset': d1})
+        assert len(sd) == 1
+        assert sd[key] == d1
+
+    def test_attr_dict(self, synced_collection, testdata):
+        sd = synced_collection
+        key = 'test'
+        d = testdata
+        sd[key] = d
+        assert len(sd) == 1
+        assert key in sd
+        assert sd[key] == d
+        assert sd.get(key) == d
+        assert sd.test == d
+        del sd.test
+        assert len(sd) == 0
+        assert key not in sd
+        d1 = testdata
+        key = 'test2'
+        sd.test2 = d1
+        assert len(sd) == 1
+        assert key in sd
+        assert sd[key] == d1
+        assert sd.get(key) == d1
+        assert sd.test2 == d1
+
+    def test_clear(self, synced_collection, testdata):
+        sd = synced_collection
+        key = 'clear'
+        d = testdata
+        sd[key] = d
+        assert len(sd) == 1
+        assert sd[key] == d
+        sd.clear()
+        assert len(sd) == 0
+
+    def test_reopen(self, synced_collection, testdata):
+        jsd = synced_collection
+        key = 'reopen'
+        d = testdata
+        jsd[key] = d
+        jsd.sync()
+        del jsd  # possibly unsafe
+        jsd2 = synced_collection
+        jsd2.load()
+        assert len(jsd2) == 1
+        assert jsd2[key] == d
+
+    def test_dfs_update(self, synced_collection, testdata):
```

Comment:
I think this name captures the purpose of the test more clearly.
```suggestion
    def test_update_recursive(self, synced_collection, testdata):
```